### PR TITLE
Create Header: swap hello name sign in button for user icon

### PR DESCRIPTION
**Pull request by Erin007**
In preparation for HOC, we are going to re-design the header to more heavily promote project creation. There are 3 main changes [spec ](https://docs.google.com/document/d/1ShWgZwO4lh1OdAn2dNbFPDCmJDowmowRk1DwMJNbGoU/edit):
1.) Swapping the "Hi, Name" button for signed in users with a user icon
2.) Adding a new create button with complementary dropdown
3.) Moving help/support documentation into a separate dropdown with ? icon 

This PR addresses item 1, with work hidden behind the page mode cookie.

If `request.cookies['pm'] == 'create_header_2018'`, you can see the new user icon in place of the signed in button. 

Pegasus using non-minified styles: 
pegasus-non-minified

Pegasus using minified styles: 
pegasus-minified

Dashboard teacher: 
teacher - dashboard

Dashboard student, not currently pairing:
dashboard - student - not pairing

Dashboard student, currently pairing: 
dashboard - student - currently pairing

No changes for signed out users: 
signed out

If you have not set the pagemode cookie, you will see the unchanged legacy signed in button.
screen shot 2018-09-20 at 11 16 27 am

Legacy, student currently pairing, dashboard: 
legacy - student - pairing

### CODE REVIEW

**Comment by Erin007** [comment]
@ryansloan Can you take peek at these screenshots and see what you think for pair programming? We didn't consider that scenario in the spec, so I made what seemed to be a reasonable assumption about what we wanted to happen.

**Comment by Erin007** [comment]
I'm pretty sure pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css isn't where I want to add the fontawesome user icon, but it worked.  Is there a better place farther upstream to add that icon to the list?

**Comment by ryansloan** [comment]
@Erin007 thanks for looping me in! The Pair Programming design looks good to me.

**Comment by breville** [comment]
@Erin007 It seems you added it to the right file.  See [here](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/views/theme_common_head_before.haml#L16) and [here](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/styles_min/README.md).

File: pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
...ont-face{font-family:'FontAwesome';src:url('/fonts/fontawesome-webfont.eot?v=4.6.3');src:url('/fonts/fontawesome-webfont.eot?#iefix&v=4.6.3') format('embedded-opentype'),url('/fonts/fontawesome-webfont.woff2?v=4.6.3') format('woff2'),url('/fonts/fontawesome-webfont.woff?v=4.6.3') format('woff'),url('/fonts/fontawesome-webfont.ttf?v=4.6.3') format('truetype'),url('/fonts/fontawesome-webfont.svg?v=4.6.3#fontawesomeregular') format('svg');font-weight:normal;font-style:normal}.fa{display:inline-block;font:normal normal normal 14px/1 FontAwesome;font-size:inherit;text-rendering:auto;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}.fa-lg{font-size:1.33333333em;line-height:.75em;vertical-align:-15%}.fa-warning:before{content:"\f071"}.fa-chevron-up:before{content:"\f077"}.fa-chevron-down:before{content:"\f078"}.fa-twitter:before{content:"\f099"}.fa-facebook:before{content:"\f09a"}.fa-caret-down:before{content:"\f0d7"}.fa-caret-up:before{content:"\f0d8"}.fa-instagram:before{content:"\f16d"}.fa-tumblr:before{content:"\f173"}.sr-only{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);border:0}.fa-user:before{content:"\f007"}
```
Comment (breville) [inline pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css:4]: Is this added in the same order as it appeared in the original file?  Might be useful to keep these sorted for doing future diffs.

Comment (Erin007) [reply pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css:4]: The diff looks really big, but the only change I made was appending `.fa-user:before{content:"\f007"}` to the whole line.

Comment (breville) [reply pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css:4]: yes, the suggestion is to not append it to the end but to insert it in the same order that it appears in the entire original font awesome source.

Comment (Erin007) [reply pegasus/sites.v3/code.org/styles_min/030-font-awesome-min.css:4]: Discussed in person, but noting here - the font-awesome-min file in styles isn't a complete super set of font-awesome-min in styles-min so this doesn't seem to be a crucial bit of house keeping.

**Pull request merged**